packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_withdraw_from_vault.test.ts

```diff
@@ -366,3 +366,387 @@ describe('loans.withdrawFromVault', () => {
+    })
+    await alice.generate(1)
+
+    // must wait until block count reaches fort canning road height
+    const blockCount = await bob.container.getBlockCount()
+    await bob.generate(fortCanningRoadHeight - blockCount)
+    await tGroup.waitForSync()
+
+    const destinationAddress = await alice.generateAddress()
+    const accountBalancesBefore = await alice.rpc.account.getAccount(destinationAddress)
+    expect(accountBalancesBefore.length).toStrictEqual(0)
+
+    const tslaLoanAmount = 2500 // loan amount = 5000 USD
+    await bob.rpc.loan.takeLoan({
+      vaultId: bobVaultId,
+      amounts: `${tslaLoanAmount}@TSLA`
+    })
+
+    const decoded = fromAddress(destinationAddress, 'regtest')
+    const script = decoded?.script as Script
+    const txn = await builder.loans.withdrawFromVault({
+      vaultId: bobVaultId,
+      to: script,
+      tokenAmount: { token: 0, amount: new BigNumber(5000) }
+    }, bobVaultScript)
+    await sendTransaction(bob.container, txn)
+
+    await bob.generate(1)
+    await tGroup.waitForSync()
+
+    const accountBalancesAfter = await alice.rpc.account.getAccount(destinationAddress)
+    expect(accountBalancesAfter.length).toStrictEqual(1)
+    expect(accountBalancesAfter[0]).toStrictEqual('5000.00000000@DFI')
+
+    const vaultAfter = await bob.rpc.loan.getVault(bobVaultId) as VaultActive
+    expect(vaultAfter.collateralAmounts).toStrictEqual(['5000.00000000@DUSD', '2.00000000@BTC'])
+    expect(vaultAfter.collateralValue).toStrictEqual(new BigNumber(15000))
+  })
+
+  it('should withdrawFromVault with 25% DFI + 25% DUSD collateral', async () => {
+    // must wait until block count reaches fort canning road height
+    const blockCount = await bob.container.getBlockCount()
+    await bob.generate(fortCanningRoadHeight - blockCount)
+    await tGroup.waitForSync()
+
+    const destinationAddress = await alice.generateAddress()
+    const accountBalancesBefore = await alice.rpc.account.getAccount(destinationAddress)
+    expect(accountBalancesBefore.length).toStrictEqual(0)
+
+    const tslaLoanAmount = 2500 // loan amount = 5000 USD
+    await bob.rpc.loan.takeLoan({
+      vaultId: bobVaultId,
+      amounts: `${tslaLoanAmount}@TSLA`
+    })
+
+    // remove dfi collateral, new total collateral = 12500 USD
+    let decoded = fromAddress(destinationAddress, 'regtest')
+    let script = decoded?.script as Script
+    let txn = await builder.loans.withdrawFromVault({
+      vaultId: bobVaultId,
+      to: script,
+      tokenAmount: { token: 0, amount: new BigNumber(2500) }
+    }, bobVaultScript)
+    await sendTransaction(bob.container, txn)
+
+    // remove dusd collateral, new total collateral = 10000 USD
+    decoded = fromAddress(destinationAddress, 'regtest')
+    script = decoded?.script as Script
+    txn = await builder.loans.withdrawFromVault({
+      vaultId: bobVaultId,
+      to: script,
+      tokenAmount: { token: 1, amount: new BigNumber(2499) }
```

Comment:
Can not withdraw 2500 DUSD here, but it can be done in the corresponding RPC test. need to check.

Reply:
1. in the RPC test, there's no block generation between two `withdrawFromVault`
2. But in dftx tests, `sendTransaction` itself has 1 block generated inside. 

Block generation increases the interest, and reduces the col ratio.
fix 1
https://github.com/DeFiCh/jellyfish/pull/1205/files/c4476a833623e417a3923ce557326be4e1e80a52#diff-e78d68f822ee4e0ce907b4038ed7d6b6f43fc0bf1288fbb3fdb22d18c673e135R682 not needed.

Reply:
Thanks for pointing out the reason, Now it can withdraw 2500 DUSD as in the RPC test.